File: apps/web/lib/types/reports.ts
// ── Shared ────────────────────────────────────────────────────────────────────

export interface ReportPagination {
  page: number
  per_page: number
  total: number
  total_pages: number
}

export interface ReportFilter {
  project_id?: string
  poslovoda_id?: string
  worker_id?: string
  material_id?: string
  activity_type?: string
  is_vtk?: boolean
  date_from?: string
  date_to?: string
  status?: string
  search?: string
  page?: number
  per_page?: number
}

// ── Građevinski dnevnik ───────────────────────────────────────────────────────

export interface GradevinskiDnevnikRow {
  report_id: string
  report_date: string
  project_id: string
  project_name: string
  project_address: string | null
  poslovoda_id: string
  poslovoda_name: string
  worker_id: string
  worker_name: string
  hours_worked: number
  notes: string | null
  status: string
}

export interface GradevinskiDnevnikSummary {
  total_hours: number
  workers_count: number
  projects_count: number
}

export interface GradevinskiDnevnikResponse {
  data: GradevinskiDnevnikRow[]
  pagination: ReportPagination
  summary: GradevinskiDnevnikSummary
}

// ── Građevinska knjiga ────────────────────────────────────────────────────────

export interface GradevinskaKnjigaRow {
  report_id: string
  report_date: string
  project_id: string
  project_name: string
  project_address: string | null
  poslovoda_id: string
  poslovoda_name: string
  activity_id: string
  material_id: string | null
  material_name: string
  material_code: string | null
  custom_material_name: string | null
  quantity: number
  unit: string
  activity_type: string
  is_vtk: boolean
  notes: string | null
  status: string
}

export interface GradevinskaKnjigaSummary {
  total_quantity: number
  activities_count: number
  vtk_count: number
  projects_count: number
}

export interface GradevinskaKnjigaResponse {
  data: GradevinskaKnjigaRow[]
  pagination: ReportPagination
  summary: GradevinskaKnjigaSummary
}

// ── Filter options ────────────────────────────────────────────────────────────

export interface FilterOptionProject {
  id: string
  name: string
  status: string
}

export interface FilterOptionPerson {
  id: string
  full_name: string
  supervisor_id?: string
}

export interface FilterOptionMaterial {
  id: string
  material_name: string
  project_id: string
}

export interface ReportFilterOptions {
  projects: FilterOptionProject[]
  poslovode: FilterOptionPerson[]
  workers: FilterOptionPerson[]
  materials: FilterOptionMaterial[]
  activity_types: string[]
  statuses: string[]
}

File: services/api/dto/reports.go
package dto

// ── Shared filter ─────────────────────────────────────────────────────────────

type ReportFilter struct {
	ProjectID    *string
	PoslovodaID  *string
	WorkerID     *string
	MaterialID   *string // knjiga: project_material_id
	ActivityType *string // knjiga
	IsVTK        *bool   // knjiga
	DateFrom     *string
	DateTo       *string
	Status       *string
	Search       *string
}

// ── Pagination ────────────────────────────────────────────────────────────────

type ReportPagination struct {
	Page       int `json:"page"`
	PerPage    int `json:"per_page"`
	Total      int `json:"total"`
	TotalPages int `json:"total_pages"`
}

// ── Građevinski dnevnik ───────────────────────────────────────────────────────

type GradevinskiDnevnikRow struct {
	ReportID        string  `json:"report_id"`
	ReportDate      string  `json:"report_date"`
	ProjectID       string  `json:"project_id"`
	ProjectName     string  `json:"project_name"`
	ProjectAddress  *string `json:"project_address"`
	PoslovodaID     string  `json:"poslovoda_id"`
	PoslovodaName   string  `json:"poslovoda_name"`
	WorkerID        string  `json:"worker_id"`
	WorkerName      string  `json:"worker_name"`
	HoursWorked     float64 `json:"hours_worked"`
	Notes           *string `json:"notes"`
	WorkDescription *string `json:"work_description"`
	WorkerRole      string  `json:"worker_role"`
	WorkerHoursID   *string `json:"worker_hours_id"` // non-null only for worker_daily_hours rows
	HasRevisions    bool    `json:"has_revisions"`
	HasComments     bool    `json:"has_comments"`
	Status          string  `json:"status"`
	Source          string  `json:"source"`
}

type GradevinskiDnevnikSummary struct {
	TotalHours    float64 `json:"total_hours"`
	WorkersCount  int     `json:"workers_count"`
	ProjectsCount int     `json:"projects_count"`
}

type GradevinskiDnevnikResponse struct {
	Data       []GradevinskiDnevnikRow   `json:"data"`
	Pagination ReportPagination          `json:"pagination"`
	Summary    GradevinskiDnevnikSummary `json:"summary"`
}

// ── Građevinska knjiga ────────────────────────────────────────────────────────

type GradevinskaKnjigaRow struct {
	ReportID           string  `json:"report_id"`
	ReportDate         string  `json:"report_date"`
	ProjectID          string  `json:"project_id"`
	ProjectName        string  `json:"project_name"`
	ProjectAddress     *string `json:"project_address"`
	PoslovodaID        string  `json:"poslovoda_id"`
	PoslovodaName      string  `json:"poslovoda_name"`
	ActivityID         string  `json:"activity_id"`
	MaterialID         *string `json:"material_id"` // project_material_id
	MaterialName       string  `json:"material_name"`
	MaterialCode       *string `json:"material_code"`
	CustomMaterialName *string `json:"custom_material_name"`
	Quantity           float64 `json:"quantity"`
	Unit               string  `json:"unit"`
	ActivityType       string  `json:"activity_type"`
	IsVTK              bool    `json:"is_vtk"`
	Notes              *string `json:"notes"`
	Status             string  `json:"status"`
}

type GradevinskaKnjigaSummary struct {
	TotalQuantity   float64 `json:"total_quantity"`
	ActivitiesCount int     `json:"activities_count"`
	VTKCount        int     `json:"vtk_count"`
	ProjectsCount   int     `json:"projects_count"`
}

type GradevinskaKnjigaResponse struct {
	Data       []GradevinskaKnjigaRow   `json:"data"`
	Pagination ReportPagination         `json:"pagination"`
	Summary    GradevinskaKnjigaSummary `json:"summary"`
}

// ── Filter options ────────────────────────────────────────────────────────────

type FilterOptionProject struct {
	ID     string `json:"id"`
	Name   string `json:"name"`
	Status string `json:"status"`
}

type FilterOptionPerson struct {
	ID           string  `json:"id"`
	FullName     string  `json:"full_name"`
	SupervisorID *string `json:"supervisor_id,omitempty"`
}

type FilterOptionMaterial struct {
	ID           string `json:"id"`
	MaterialName string `json:"material_name"`
	ProjectID    string `json:"project_id"`
}

type ReportFilterOptions struct {
	Projects      []FilterOptionProject  `json:"projects"`
	Poslovode     []FilterOptionPerson   `json:"poslovode"`
	Workers       []FilterOptionPerson   `json:"workers"`
	Materials     []FilterOptionMaterial `json:"materials"`
	ActivityTypes []string               `json:"activity_types"`
	Statuses      []string               `json:"statuses"`
}
